crates/vault-registry/src/lib.rs

```diff
@@ -872,6 +872,9 @@ impl<T: Config> Pallet<T> {
         // will fail if free_balance is insufficient
         amount.lock_on(&vault_id.account_id)?;
 
+        // withdraw first such that past rewards don't get changed by this deposit
+        ext::fee::withdraw_all_vault_rewards::<T>(vault_id)?;
```

Comment:
Why is this necessary? Nomination does the same thing to prevent new nominators earning past rewards but I'm not sure why this matters here

Reply:
It's a change brought to light by my randomized tests, so I was quite happy about that :)

To see why this is necessary, consider this scenario:
- vault deposits 1 dot collateral
- nominator deposits 1 dot collateral
- 1 INTR rewards are distributed. Both vault and nominator should get 50% (ignoring commission). But we don't withdraw yet.
- vault deposits 1 additional dot
- Now we withdraw from the reward pool to the staking pool. With the current stakes, vault gets 2/3rd, nominator gets 1/3rd --> wrong!

Reply:
Good catch! Makes sense to me, thanks for the example